cpp/src/arrow/compute/kernels/ree_util_internal.cc

```diff
@@ -62,7 +63,7 @@ Result<std::shared_ptr<ArrayData>> PreallocateValuesArray(
   std::vector<std::shared_ptr<Buffer>> values_data_buffers;
   std::shared_ptr<Buffer> validity_buffer = NULLPTR;
   if (has_validity_buffer) {
-    ARROW_ASSIGN_OR_RAISE(validity_buffer, AllocateBitmap(length, pool));
+    ARROW_ASSIGN_OR_RAISE(validity_buffer, AllocateEmptyBitmap(length, pool));
```

Comment:
Do we have to pay the cost of initializing the buffers even though every value will be overwritten by the code that uses this buffer?

Reply:
We do this in other places already. Initializing a buffer that's just been allocated (and that will be accessed just after) should be reasonably cheap.

I would welcome another solution, but we don't want to disable the Valgrid checks as they're important.